libminifi/include/utils/file/FileUtils.h

```diff
@@ -112,31 +112,28 @@ class FileUtils {
 #endif
   }
 
-  static std::string create_temp_directory(char *format) {
+  static std::string create_temp_directory(char * const format) {
```

Comment:
What is the purpose of using a const pointer (not pointer to const) here as a function argument? It gives no extra guarantees to the caller.

Reply:
It belongs to the definition, not the declaration. It doesn't matter for the caller but provides extra checks for the definition. I tend to use `const` whenever it's possible and I don't know a reason not to.

ref1: this slide: https://youtu.be/iz5Qx18H6lg?t=293
ref2: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable

Reply:
I understand your intent, but it can be confusing for users when looking at the function signature. I agree with making as many things const as possible, but this, in my opinion, reduces readability of the API without having much benefit.
None of your references specifically address the case of having const pointers in function signatures, they are just speaking generically about immutable objects - and as I said, in the generic case, I agree.
Neither do I know of any major projects that would use const pointers on their API.

Reply:
@szaszm my bad, your second reference actually addresses the issue, and spells out the reasons I give not to use it in this case:
```Exception Function arguments are rarely mutated, but also rarely declared const. To avoid confusion and lots of false positives, don’t enforce this rule for function arguments.```

Reply:
Pointers are objects. My interpretation of the exception is that the issue is too minor to be enforced by static analyzers, and is commonly not followed, hence the "false positives". In other words, my interpretation of the exception doesn't suggest that function argument objects shouldn't be declared `const`.

I can see the point behind the readability argument, since `const` is present in the argument list, yet, the string behind the pointer is mutated. I think the risk of misunderstanding is minor given that we expect people who view the source to know the language well enough to know what `const` applies to. With this assumption, IMO the value gained by extra static analysis is greater than the value of lost readability.

I couldn't find any clear guidelines on the `const`-ness of argument pointers. I'm willing to remove `const` if you insist on it.

Reply:
Our core code is expected to be written by people very familiar with the language. We strive to create an ecosystem where extensions can be written (our API can be used) safely even by people without great familiarity with the language.
This change introduces a risk of misunderstanding by someone less familiar with a language (or someone familiar with it, just running low on caffeine), without bringing comparable benefits.
Please revert it.

Reply:
reverted